### PR TITLE
Add Issue and PR links to Roadmap.md

### PR DESCRIPTION
**Pull request by akestner**
Issue #, if available: 
#485 (partially)

**Description of changes:**
This PR updates the `ROADMAP.md` doc to include links to the relevant Issues and PRs for each feature listed. In the cases where multiple PRs were needed to close out a feature, we've listed the PR that first introduced the code needed for the new feature (as opposed to continued refinements/enhancements).

This is the first step in migrating the project to the GitHub [Product Roadmap](https://github.com/awslabs/karpenter/projects/3).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by njtran — COMMENTED** [review]
I love this idea. For some issues, especially ones with higher t-shirt sizes, we end up having multiple PRs for the same issue. Maybe we could simply just link the issues, where the PRs associated will show up in that thread naturally. 

Additionally, could you use [#322](link) syntax so people can click on the links?

**Comment by ellistarn** [comment]
I might just hold off on this and whole-hog migrate to the new system. Unless you think we can't get there by v0.3

**Comment by akestner** [comment]
Happy to close this in favor of a big red PR that deletes this whole table from the ROADMAP.md doc. Part of the work to migrate was to link up Issues with PRs and thought surfacing this could be a useful intermediate step at which to get some feedback

**Comment by akestner** [comment]
Totally agree that the Issues and PRs should be links as per usual. Part of the reason to open this as a draft was to see if GitHub was clever enough to automatically turn the `#123` issue numbers in the doc into links.  My forked repo obviously doesn't have any issues or PRs

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
-Issue #, if available:
+**Issue # (if avilable):**
```
Comment (ellistarn) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: avil able. Also, folks will need to add the hash themselves, and Issue #: #123 stutters. Perhaps we can just do 
```
**Issue #**
```

**Comment by ellistarn** [comment]
> Happy to close this in favor of a big red PR that deletes this whole table from the ROADMAP.md doc. Part of the work to migrate was to link up Issues with PRs and thought surfacing this could be a useful intermediate step at which to get some feedback

Not fussed either way. Happy to approve, you can decide if/when to merge.

**Comment by njtran** [comment]
>Part of the reason to open this as a draft was to see if GitHub was clever enough to automatically turn the #123 issue numbers in the doc into links

Do you know if it would work when this is merged? 

Additionally, per my earlier point of linking each of the issues with markdown syntax: 
Since not all read the enriched compiled version of markdown files, using markdown link syntax might be tough to read. 

Like Ellis, happy to approve.

**Comment by akestner** [comment]
Closing until we have completely migrated the roadmap over

**Pull request closed without merging**